lib/phoenix_live_view/html_engine.ex

```diff
@@ -292,12 +318,39 @@ defmodule Phoenix.LiveView.HTMLEngine do
     |> update_subengine(:handle_expr, ["=", ast])
   end
 
+  # Slot
+
+  defp handle_token({:tag_open, ":" <> _, _attrs, _tag_meta} = token, state) do
+    state
+    |> push_tag(token)
+    |> push_substate_to_stack()
+  end
+
+  defp handle_token({:tag_close, ":" <> slot_name, _tag_close_meta} = token, state) do
+    {{:tag_open, _name, attrs, %{line: line} = tag_meta}, state} = pop_tag!(state, token)
+
+    {let, assigns} = handle_component_attrs(attrs, state.file)
+    clauses = build_component_clauses(let, state)
+
+    ast =
+      quote line: line do
+        Phoenix.LiveView.Helpers.slot(:inner_block, unquote(assigns), do: unquote(clauses))
```

Comment:
Instead of `:inner_block`, this should be the name of the `slot`, because that's what we will use to look for in the changed struct.